core/transport/base.py

```diff
@@ -0,0 +1,95 @@
+from typing import List, Callable, TypeVar, Dict, Any, Optional
+
+
+class AgentGatewayBase:
+    _on_service_callback: Optional[Callable]
+    _on_channel_callback: Optional[Callable]
+
+    def __init__(self, on_service_callback: Optional[Callable] = None,
+                 on_channel_callback: Optional[Callable] = None, *args, **kwargs):
+
+        super(AgentGatewayBase, self).__init__(*args, **kwargs)
+        self._on_service_callback = on_service_callback
+        self._on_channel_callback = on_channel_callback
+
+    @property
+    def on_service_callback(self):
+        return self._on_service_callback
+
+    @on_service_callback.setter
+    def on_service_callback(self, callback: Callable):
+        self._on_service_callback = callback
+
+    @property
+    def on_channel_callback(self):
+        return self._on_channel_callback
+
+    @on_channel_callback.setter
+    def on_channel_callback(self, callback: Callable):
+        self._on_channel_callback = callback
+
+    async def send_to_service(self, service: str, dialog: Dict) -> None:
+        raise NotImplementedError
+
+    async def send_to_channel(self, channel_id: str, user_id: str, response: str) -> None:
+        raise NotImplementedError
+
+
+TAgentGateway = TypeVar('TAgentGateway', bound=AgentGatewayBase)
```

Comment:
it's better to put all types to the bottom of the file

Reply:
In this file it is ok. Moved.